derive/tests/pubsub-macros.rs

```diff
@@ -0,0 +1,81 @@
+extern crate serde_json;
+extern crate jsonrpc_core;
+extern crate jsonrpc_pubsub;
+#[macro_use]
+extern crate jsonrpc_derive;
+
+use std::sync::Arc;
+use jsonrpc_core::futures::sync::mpsc;
+use jsonrpc_pubsub::{PubSubHandler, SubscriptionId, Session, PubSubMetadata};
+use jsonrpc_pubsub::typed::Subscriber;
+
+pub enum MyError {}
+impl From<MyError> for jsonrpc_core::Error {
+	fn from(_e: MyError) -> Self {
+		unreachable!()
+	}
+}
+
+type Result<T> = ::std::result::Result<T, MyError>;
+
+#[rpc(pubsub, name = "hello")]
```

Comment:
Does this mean that each pubsub now has to be a separate trait?

Reply:
Yes at the moment it does. It was a way to keep the grouping of the sub/unsub method pair which is done with a `{ }` block in the existing macro. I'm not 100% happy with it (not least beause in substrate it means copying generic soup for the separate traits). Any better suggestions to group the sub/unsub pair in the same trait?

Reply:
Yeah, I'm not happy with the block either maybe let's repeat the subscription name?

Alternatively let's have only one method per subscription and just pass an enum:
```
enum Subscription<SinkValue, ParamsTuple> {
  Subscribe { meta, subscriber, params },
  Unsubscribe { meta, subscriber, params },
}
```
What do you think? That might complicate the attribute significanlty (cause you need to provide name, subscribe, unsubscribe and potential aliases)

Reply:
Repeating the subscription name seems like an easy solution. However I hadn't thought of the enum idea - I will at least try it out to see how it looks in practice. Job for next week.

Reply:
In the end I went for repeating the subscription name, since it was easier to implement with what I had already done.